internal/message/subtrie.go

```diff
@@ -42,6 +42,7 @@ type Trie struct {
 	sync.RWMutex
 	root  *node // The root node of the tree.
 	count int   // Number of subscriptions in the trie.
+	topicMatchingType string
```

Comment:
Let's change this struct field to `lookup func(Ssid, *Subscribers, *node, func(s Subscriber) bool)`. This way in the constructor function we can simply assign the appropriate function pointer instead of doing a string matching every time.